src/v2/view-builder/views/google-authenticator/EnrollAuthenticatorGoogleAuthenticatorView.js

```diff
@@ -0,0 +1,98 @@
+import { loc, createButton } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import EnrollGoogleAuthenticatorBarcodeView from './EnrollGoogleAuthenticatorBarcodeView';
+import EnrollAuthenticatorMannuallySetupView from './EnrollAuthenticatorMannuallySetupView';
+
+const Body = BaseForm.extend({
+  title () {
+    return loc('oie.enroll.google_authenticator.setup.title', 'login');
+  },
+
+  subtitle () {
+    return loc('oie.enroll.google_authenticator.enterCode.title', 'login');
+  },
+
+  save () {
+    return loc('mfa.challenge.verify', 'login');
+  },
+
+  className: 'oie-enroll-google-authenticator',
+
+  modelEvents: {
+    'triggerBarcodeView': '_triggerBarcodeView'
+  },
+
+  _triggerBarcodeView () {
+    this.$('.oie-enroll-google-authenticator-barcode').hide();
+    this._showManuallySetupView();
+  },
+
+  _showVerifyCodePage () {
+    this.$('.oie-enroll-google-authenticator-barcode').hide();
+    this.$('.google-authenticator-next').hide();
+    this._hideManuallySetupView();
+    this.$('.o-form-label-top').show();
+    this.$('.o-form-button-bar').show();
+  },
+
+  _hideVerifyCodePage () {
+    this.$('.o-form-label-top').hide();
```

Comment:
Try adding a class to the schema of this input and hide that instead ?

Reply:
the way we are hiding individual containers looks pretty weak to me.. For eg if we add a new container at global level then the we have to come here and update the code to toggle it..

I would suggest to move show/hide (common logic) to BaseForm and create a generalized function that any view can overwrite.. 

For eg: 
 https://github.com/okta/okta-signin-widget/blob/master/src/v2/view-builder/internals/BaseForm.js#L37
 
 @haishengwu-okta Do you have any inputs how to handle this ??

Reply:
Hi @magizh-okta this is the Button that added in the BaseForm by default.